gradio/components/image.py

```diff
@@ -47,15 +45,16 @@ def __init__(
         self,
         value: str | _Image.Image | np.ndarray | None = None,
         *,
-        shape: tuple[int, int] | None = None,
         height: int | None = None,
         width: int | None = None,
         image_mode: Literal[
             "1", "L", "P", "RGB", "RGBA", "CMYK", "YCbCr", "LAB", "HSV", "I", "F"
         ] = "RGB",
-        invert_colors: bool = False,
-        source: Literal["upload", "webcam", "canvas"] = "upload",
-        tool: Literal["editor", "select", "sketch", "color-sketch"] | None = None,
+        sources: list[Literal["upload", "webcam", "clipboard"]] = [
+            "upload",
+            "webcam",
+            "clipboard",
+        ],  # ruff: noqa: B006
```

Comment:
ruff doesn't like lists as a default argument

Reply:
Mutable default arguments are actually not good practice because they get shared across all instances of the class. 

I think we can:
* Set default to None and set sources to ["upload", "webcam", "clipboard"] if it's None
* Set the type to `Iterable[Literal` and use a tuple as the default argument.

Second might be nicer.

Reply:
cool

Reply:
We use the first approach in most other places, so maybe we should go with that. But don't feel too strongly.